### PR TITLE
Wave Transmission Model added

### PR DESCRIPTION
**Pull request by behdadebsh**
Wave Trasmission Model added to aether library with all the required bindings and compatible with updated perfusion code and growing. Merged from alysclark/lungsim-wave_transmission_model branch.

### CODE REVIEW

**Comment by alysclark** [comment]
Hi @behdadebsh Can you confirm that you are now able to run the wave transmission model and get exactly the same results as when running on the other branch. Otherwise this pull request may be premature and its better to close it and aim to do a pull request when you have this fixed.

File: src/lib/capillaryflow.f90 [hunk 1]
```diff
@@ -283,7 +283,7 @@ subroutine evaluate_ladder(ne,NonZeros,MatrixSize,submatrixsize,ngen,&
               '(I6,X,3(F9.2,X),I6,X,4(F8.2,X),4(F8.5,X),&
          2(F10.2,X),3(F8.4,X),I6,X,2(F10.5,X),2(F8.4,X),&
          (F10.2,X))')&
-         ne,x,y,z,gen,Pin,Pin_sheet,Pout_sheet,Pout,Qtot*1.d9,&
+         elem_cnct(-1,1,ne),x,y,z,gen,Pin,Pin_sheet,Pout_sheet,Pout,Qtot*1.d9,&
```
Comment (alysclark) [inline src/lib/capillaryflow.f90]: This change is not correct, the line should remain "ne"

File: src/lib/capillaryflow.f90 [hunk 2]
```diff
@@ -306,7 +306,7 @@ subroutine evaluate_ladder(ne,NonZeros,MatrixSize,submatrixsize,ngen,&
          WRITE(10,&
         '(I6,X,3(F9.2,X),I6,X,4(F8.2,X),4(F8.5,X),&
         2(F10.2,X),3(F8.4,X),I6,X,2(F10.5,X),3(F8.4,X))')&
-         ne,x,y,z,gen,Pin,Pin_sheet,Pout_sheet,Pout,Qtot*1.d9,&
+         elem_cnct(-1,1,ne),x,y,z,gen,Pin,Pin_sheet,Pout_sheet,Pout,Qtot*1.d9,&
```
Comment (alysclark) [inline src/lib/capillaryflow.f90]: again , this should remain "ne"

File: src/lib/capillaryflow.f90 [hunk 3]
```diff
@@ -328,7 +328,7 @@ subroutine evaluate_ladder(ne,NonZeros,MatrixSize,submatrixsize,ngen,&
 ! Rtot=9 Pa/mm^3 | Blood_vol=10 mm^3| sheet_area= 11 mm^2 | ave_TT=12 s |ave_H=13 um |Ppl=14 Pa
           WRITE(20,&
         '(I6,X,5(F9.2,X),2(F8.5,X),F10.2,X,F8.4,X,F10.4,X,F10.3,X,F8.4,X,F9.4,X)') &
-         ne,x,y,z,Pin,Pout,Q01_mthrees*1.d9,Qtot*1.d9,Rtot/1000.d0**3.d0,&
+         elem_cnct(-1,1,ne),x,y,z,Pin,Pout,Q01_mthrees*1.d9,Qtot*1.d9,Rtot/1000.d0**3.d0,&
```
Comment (alysclark) [inline src/lib/capillaryflow.f90]: ne

File: src/lib/capillaryflow.f90 [hunk 4]
```diff
@@ -884,11 +885,10 @@ subroutine cap_specific_parameters(ne,Ppl,alpha_c,area_scale,length_scale,l_a,ra
 
 !    --ARTERIOLE AND VENULE PROPERTIES--
 !###  APPARENT BLOOD VISCOSITY:
-!...  Stepping down linearly with each generation from Fung's
-!...  estimate at 45% hematocrit (4e-3Pa.s) to his estimate at 30% hematocrit
+!...  Stepping down linearly with each generation from the larger vessels to (4e-3Pa.s) to his estimate at 30% hematocrit
 !...  (1.92e-3Pa.s). (Biomechanics: Circulation)
       DO i=1,cap_param%num_symm_gen
-         mu_app(i)=(4.0e-3_dp-(i-1)*(4.0e-3_dp-1.92e-3_dp)/(cap_param%num_symm_gen-1));
+         mu_app(i)=(3.360e-3_dp-(i-1)*(3.36e-3_dp-1.92e-3_dp)/(cap_param%num_symm_gen-1));
```
Comment (alysclark) [inline src/lib/capillaryflow.f90]: This formula is incorrect in terms of the steady state model equations. Should be as original in both OR this needs to be a parameter in the model

File: src/lib/geometry.f90 [hunk 5]
```diff
@@ -226,7 +226,7 @@ subroutine add_matching_mesh()
     call enter_exit(sub_name,1)
     !Ultimately offset should be an input argument
     offset(1)=0.0_dp
-    offset(2)=1e-6_dp
+    offset(2)=0.0_dp
```
Comment (alysclark) [inline src/lib/geometry.f90]: This is a case where what was in the repository was correct. Please revert back

File: src/lib/geometry.f90 [hunk 6]
```diff
@@ -284,10 +284,10 @@ subroutine add_matching_mesh()
           elem_nodes(2,ne)=np_map(elem_nodes(1,ne_m))
           elem_cnct(-1,0,ne)=elem_cnct(1,0,ne_m)
           elem_cnct(1,0,ne)=elem_cnct(-1,0,ne_m)
-          do n=1,elem_cnct(1,0,ne)
+          do n=1,elem_cnct(-1,0,ne)
```
Comment (alysclark) [inline src/lib/geometry.f90]: Is there good reason to change this? What is in the current repository shouldnt need to be changed?

**Comment by behdadebsh** [comment]
Hi @alysclark . I can confirm that this branch is now functional in wave_trasmission model showing the same results as the other branch. I also tested the grow_tree and perfusion codes and no conflicts were noticed.

File: src/lib/indices.f90 [hunk 7]
```diff
@@ -20,7 +20,7 @@ module indices
      nj_conc2
   ! indices for elem_field
   integer ::num_ne,ne_radius,ne_length,ne_vol,&
-      ne_resist,ne_t_resist,ne_Vdot,ne_Vdot0,ne_a_A,&
+      ne_resist,ne_t_resist,ne_flow,ne_flow0,ne_Vdot,ne_Vdot0,ne_a_A,&
```
Comment (alysclark) [inline src/lib/indices.f90]: ne_flow has been replaced in the code to give more meaningful field names. Shoudl not be reintroduced.  These are ne_Vdot for ventilation and ne_Qdot for perfusion

File: src/lib/indices.f90 [hunk 8]
```diff
@@ -210,6 +211,8 @@ subroutine ventilation_indices
     ne_vol=3 !volume
     ne_resist=4 !resistance of airway
     ne_t_resist=5
+    ne_flow=6
```
Comment (alysclark) [inline src/lib/indices.f90]: please remove this change it is not correct

File: src/lib/indices.f90 [hunk 9]
```diff
@@ -276,6 +279,8 @@ subroutine perfusion_indices
     ne_length=4!length of an elevent
     ne_radius_in0=5!unstrained radius into an element
     ne_radius_out0=6!unstrained radius out of an element
+    ne_flow=7
```
Comment (alysclark) [inline src/lib/indices.f90]: 1) ne_flow should be neQdot throughout, 2) why have you defined ne_resist twice?

File: src/lib/pressure_resistance_flow.f90 [hunk 10]
```diff
@@ -72,7 +72,13 @@ subroutine evaluate_prq(mesh_type,grav_dirn,grav_factor,bc_type,inlet_bc,outlet_
     !pressure (at inlet and outlets)
     !flow (flow at inlet pressure at outlet).
 
-vessel_type='elastic_g0_beta'
+
+!vessel_type='elastic_g0_beta'
+
+!mesh_type='full_plus_ladder'
```
Comment (alysclark) [inline src/lib/pressure_resistance_flow.f90]: Why have commented out lines been added. No need to do this!

File: src/lib/pressure_resistance_flow.f90 [hunk 11]
```diff
@@ -103,6 +109,10 @@ subroutine evaluate_prq(mesh_type,grav_dirn,grav_factor,bc_type,inlet_bc,outlet_
      call exit(0)
 endif
 
+
+!grav_dirn=2
+!grav_factor=1.0_dp
+
```
Comment (alysclark) [inline src/lib/pressure_resistance_flow.f90]: Again no need to add commented out lines.

File: src/lib/pressure_resistance_flow.f90 [hunk 12]
```diff
@@ -119,6 +129,9 @@ subroutine evaluate_prq(mesh_type,grav_dirn,grav_factor,bc_type,inlet_bc,outlet_
 if(bc_type.eq.'pressure')then
     inletbc=inlet_bc
     outletbc=outlet_bc
+    !inletbc=2038.0_dp!15.0_dp*133.0_dp!2266.0_dp
```
Comment (alysclark) [inline src/lib/pressure_resistance_flow.f90]: delete thes e lines

**Comment by alysclark** [comment]
Superceded by new pull request #85

**Pull request closed without merging**